src/opcodes/arithmetic.cairo

```diff
@@ -93,6 +96,9 @@ pub fn opcode_bool_or(ref engine: Engine) -> Result<(), felt252> {
 pub fn opcode_numequal(ref engine: Engine) -> Result<(), felt252> {
     let a = engine.dstack.pop_int()?;
     let b = engine.dstack.pop_int()?;
+    if a > MAX_INT32.into() || a < MIN_INT32.into() || b > MAX_INT32.into() || b < MIN_INT32.into() {
```

Comment:
It would be better to have this range check as part of `pop_int` and `peek_int` so other arithmetic operations don't need to implement the same checks. Thanks